xml_converter/src/xml_converter.cpp

```diff
@@ -144,20 +152,29 @@ void process_data(
 
     // This is a special output path used for burrito internal use that splits
     // the guildpoint protobins by map id.
-    string output_split_guildpoint_dir) {
+    string output_split_guildpoint_dir,
+    bool allow_duplicates) {
     // All of the loaded pois and categories
     vector<Parseable*> parsed_pois;
     map<string, Category> marker_categories;
+    set<string> top_level_categories;
+    vector<string> duplicate_categories;
 
     // Read in all the xml taco markerpacks
     auto begin = chrono::high_resolution_clock::now();
     for (size_t i = 0; i < input_taco_paths.size(); i++) {
         cout << "Loading taco pack " << input_taco_paths[i] << endl;
 
-        read_taco_directory(
+        set<string> category_names = read_taco_directory(
             input_taco_paths[i],
             &marker_categories,
             &parsed_pois);
+        for (string category_name : category_names) {
+            if (find(top_level_categories.begin(), top_level_categories.end(), category_name) != top_level_categories.end())
```

Comment:
Is this the right function? I would think it would be `top_level_categories.find(category_name)`. Where is this `find()` function from and why do you need to pass `begin()` and `end()` iterators into it?

Reply:
I'm guessing algorhtim's find is an o(n) linear search, which is much worse than set.finds(log(n)). Iirc set uses a tree not a hash table behind the scenes.